lib/redix/pubsub/connection.ex

```diff
@@ -230,6 +259,19 @@ defmodule Redix.PubSub.Connection do
     end
   end
 
+  def connected({:call, from}, :ping, data) do
+    random_string = :crypto.strong_rand_bytes(12) |> Base.encode64()
+    pipeline = [["PING", "ping:#{random_string}"]]
+    data.transport.send(data.socket, Enum.map(pipeline, &Protocol.pack/1))
+
+    Process.send_after(self(), {:ping_callback, random_string}, 5000)
```

Comment:
Don't use `send_after/3` here. Instead, use `:gen_statem` timeout actions. Return a timeout action from the return value of this function and handle the timeout instead of the message later on. Cancel the timeout if you get a ping with `:cancel`. The `:gen_statem` docs are the best resource for this 🙃